android/src/toga_android/fonts.py

```diff
@@ -96,6 +99,7 @@ def typeface(self, *, default=Typeface.DEFAULT):
     def size(self, *, default=None):
         """Return the font size in physical pixels."""
         context = MainActivity.singletonThis
+        base_size = 14
```

Comment:
This size will need to be scaled in a similar way to the `TypedValue.applyDimension` code below. Otherwise, the text will be too small on high DPI devices (which is virtually all of them). For example, if I edit `examples/font_size` to use the keywords:

![Screenshot_20250414_215333](https://github.com/user-attachments/assets/04f6f6d7-1df4-46dd-a371-c286803f4c08)

As mentioned at https://github.com/beeware/toga/issues/1814#issuecomment-1480800981, "medium" should be the same as the default for everything except the TextInput, where the default should be larger.

Please make a similar visual check on as many other platforms as you can, and let me know which ones you've checked. Windows in high DPI mode is the most likely one to have a problem.